crates/sncast/src/starknet_commands/script.rs

```diff
@@ -37,18 +37,87 @@ use scarb_artifacts::ScarbCommand;
 use sncast::helpers::scarb_utils::{
     get_package_metadata, get_scarb_manifest, get_scarb_metadata_with_deps, CastConfig,
 };
-use sncast::response::structs::ScriptResponse;
+use sncast::response::print::{print_command_result, OutputFormat, OutputValue};
+use sncast::response::structs::{CommandResponse, ScriptResponse};
+use sncast::NumbersFormat;
 use starknet::accounts::Account;
 use starknet::core::types::{BlockId, BlockTag::Pending, FieldElement};
 use starknet::providers::jsonrpc::HttpTransport;
 use starknet::providers::JsonRpcClient;
 use tokio::runtime::Runtime;
 
+#[derive(Copy, Clone, Debug, Default, Eq, PartialEq, Ord, PartialOrd)]
+pub enum Verbosity {
+    /// Silence all script output except for errors
+    Quiet,
+
+    /// Default verbosity level
+    #[default]
+    Normal,
+}
+
+#[derive(Debug)]
+pub struct UI {
+    verbosity: Verbosity,
+    numbers_format: NumbersFormat,
+    output_format: OutputFormat,
+}
+
+impl UI {
+    pub fn new(is_quiet: bool, numbers_format: NumbersFormat, output_format: OutputFormat) -> Self {
+        let verbosity = if is_quiet {
+            Verbosity::Quiet
+        } else {
+            Verbosity::Normal
+        };
+
+        Self {
+            verbosity,
+            numbers_format,
+            output_format,
+        }
+    }
+
+    pub fn is_quiet(&self) -> bool {
+        self.verbosity < Verbosity::Normal
+    }
+
+    pub fn is_json(&self) -> bool {
+        self.output_format == OutputFormat::Json
+    }
+
+    pub fn print_subcommand_response<T: CommandResponse>(
+        &self,
+        command: &str,
+        response: T,
+    ) -> Result<()> {
+        if self.verbosity >= Verbosity::Normal {
+            let header = (
+                String::from("script_subcommand"),
+                OutputValue::String(command.to_string()),
+            );
+            print_command_result(
+                header,
```

Comment:
I don't like passing the header, you need to add a semantics for header because of that. Would be better to just return the data and print it.

Reply:
For example here, do you actually want to put subcommand errors to std_error? because I am not sure about that imo those are not errors in terms of cast behaviour.